client/app/bundles/course/material/folders/handles.ts

```diff
@@ -3,11 +3,25 @@ import { getIdFromUnknown } from 'utilities';
 import CourseAPI from 'api/course';
 import { CrumbPath, DataHandle } from 'lib/hooks/router/dynamicNest';
 
+export const loadDefaultMaterialId = async (): Promise<number> => {
+  const {
+    data: { id },
+  } = await CourseAPI.materials.fetchDefault();
+  return id;
+};
+
 const getFolderTitle = async (
-  courseUrl: string,
+  courseId: string,
   folderId: number,
 ): Promise<CrumbPath> => {
-  const { data } = await CourseAPI.folders.fetch(folderId);
+  const courseUrl = `/courses/${courseId}`;
+  let data;
+  try {
+    ({ data } = await CourseAPI.folders.fetch(folderId));
+  } catch (error) {
+    const defaultMaterialId = await loadDefaultMaterialId();
+    ({ data } = await CourseAPI.folders.fetch(defaultMaterialId));
```

Comment:
You should align your backend API so you don't need to make 3 API calls just to get the root folder (2 calls are sufficient).  
1. fetch invalid folder payload (throws error)
2. ~~fetch valid root folderId~~
3. fetch valid root folder payload